DotNet/DataAcquisition.Domain/Application/Models/ResourceTypeModel.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Text.Json.Serialization;
+
+namespace LantanaGroup.Link.DataAcquisition.Domain.Application.Models;
+
+[JsonConverter(typeof(JsonStringEnumConverter))]
+public enum ResourceTypeModel
+{
+    PatientList,
+    Encounter,
+    Condition,
+    MedicationRequest,
+    Observation,
+    Procedure,
+    ServiceRequest,
+    Coverage,
+    MedicationAdminisitration,
+    Location
+}
```

Comment:
Not sure where this is used yet but may need to add Device, Diagnostic Report and Medication to this.